### ISSUE TITLE
False Negative on reporting a bug

### ISSUE THREAD

**Issue by lapistano** [scrutinizer-ci/php-analyzer#651]
**Code**:
```php
 public function do()
    {
        $config = doMore();
        $this->validateParsedConfiguration($config, 'Some message.');
        $configuration = $this->processor->processConfiguration($this, $config);
        $this->validateParsedConfiguration($configuration, 'Another message');

        return $configuration;
    }
```
**Message**:
```
It seems like $config defined by doMore() on line xy can also be of type string; however, Processor:processConfiguration() does only seem to accept array, maybe add an additional type check? 
```
**Explaination**
Both $config and $configuration are validated to be an array in validateParsedConfiguration(). In case they are not parsed into an array correctly validateParsedConfiguration() throws a parser exception.
This is why I think the bug highlighting is a false negative.

see https://scrutinizer-ci.com/g/libgraviton/deploy-scripts/inspections/56bd6f09-799d-4936-82a7-c3e9eaccb791/issues/files/src/Configuration.php?status=new&selectedLabels%5B0%5D=9&orderField=path&order=asc&issueId=4b328b04-502f-49d1-a1b3-0f8b37f282d5
